compiler/resolution/postFold.cpp

```diff
@@ -644,6 +644,15 @@ static Expr* postFoldPrimop(CallExpr* call) {
 
     call->replace(retval);
 
+  } else if (call->isPrimitive(PRIM_REAL64_AS_UINT64)) {
+    Expr* realArg = call->get(1);
+    Immediate* realVal = getSymbolImmediate(toSymExpr(realArg)->symbol());
+    Immediate uintVal;
+    uintVal.const_kind = NUM_KIND_UINT;
+    uintVal.num_index = INT_SIZE_64;
+    memcpy(&(uintVal.v_uint64), &(realVal->v_float64), 8);
+    retval = new SymExpr(new_ImmediateSymbol(&uintVal));
+    call->replace(retval);
```

Comment:
IMO this would be simpler to implement like this:

```
    Expr* realArg = call->get(1);
    Immediate* realVal = getSymbolImmediate(toSymExpr(realArg)->symbol());
    double f;
    uint64_t u;
    INT_ASSERT(sizeof(f) == sizeof(u));
    memcpy(&u, &f, sizeof(f));
    retval = new SymExpr(new_UIntSymbol(u, INT_SIZE_64));
    call->replace(retval);
```

* hides details of Immediate fields
* uses new_UIntSymbol which is more typical

Either way you still have the other 3 new primitives to implement

Reply:
> Either way you still have the other 3 new primitives to implement

Hence the "draft" marker, as noted on https://github.com/chapel-lang/chapel/issues/14140#issuecomment-1245990022

Reply:
Getting back to this after being distracted the past 3 months, I like your approach, but think it would be even better if it initialized `f`.  XD